### PR TITLE
[MOB-2373] Call request callbacks on task completion

### PR DESCRIPTION
**Pull request by vbabenkoru**
## 🔹 Jira Ticket(s) if any

* [MOB-2373](https://iterable.atlassian.net/browse/MOB-2373)

## ✏️ Description

Call original request callbacks when a task is completed.

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableTaskRunner.java [hunk 1]
```diff
@@ -67,15 +90,33 @@ private boolean processNextTask() {
...
+            TaskResult result = TaskResult.FAILURE;
             try {
                 IterableApiRequest request = IterableApiRequest.fromJSON(new JSONObject(task.data), null, null);
-                IterableApiResponse response = IterableRequestTask.executeApiRequest(request);
+                response = IterableRequestTask.executeApiRequest(request);
             } catch (Exception e) {
                 IterableLogger.e(TAG, "Error while processing request task", e);
             }
+            if (response != null) {
+                result = response.success ? TaskResult.SUCCESS : TaskResult.FAILURE;
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableTaskRunner.java:102]: Wow!!! I didn't know we have this feature in java!

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableTaskRunner.java [hunk 2]
```diff
@@ -67,15 +90,33 @@ private boolean processNextTask() {
...
         return false;
     }
+
+    @WorkerThread
+    private void callTaskCompletedListeners(final String taskId, final TaskResult result, final IterableApiResponse response) {
+        for (final TaskCompletedListener listener : taskCompletedListeners) {
+            new Handler(Looper.getMainLooper()).post(new Runnable() {
+                @Override
+                public void run() {
+                    listener.onTaskCompleted(taskId, result, response);
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableTaskRunner.java:117]: reiterating form previous PR, if the delete task should happen here regardless of success or failure.

File: iterableapi/src/main/java/com/iterable/iterableapi/OfflineRequestProcessor.java [hunk 3]
```diff
@@ -59,4 +63,18 @@ void scheduleTask(IterableApiRequest request, @Nullable IterableHelper.SuccessHa
...
         failureCallbackMap.put(taskId, onFailure);
     }
+
+    @MainThread
+    @Override
+    public void onTaskCompleted(String taskId, IterableTaskRunner.TaskResult result, IterableApiResponse response) {
+        IterableHelper.SuccessHandler onSuccess = successCallbackMap.get(taskId);
+        IterableHelper.FailureHandler onFailure = failureCallbackMap.get(taskId);
+        successCallbackMap.remove(taskId);
+        failureCallbackMap.remove(taskId);
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/OfflineRequestProcessor.java:73]: or should the delete happen here? 🤷‍♂️

**Review by Ayyanchira — COMMENTED** [review]
Amazing!!! 🤩

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableTaskRunner.java (hunk 2 above)
Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableTaskRunner.java:117]: Responded in the previous PR.

**Review by Ayyanchira — APPROVED** [review]

**Pull request merged**